### DISCUSSION TITLE
Logging for Disconnecting Events

### DISCUSSION THREAD

**Discussion by Nicky9319** [miguelgrinberg/python-socketio#1405] [Q&A]
I am using the websockets to send and receive a lot of data maybe in hundreds of MB and I know it is not ideal which I would change in near future.
Currently when sending this amount of data there are sometimes disconnects in between and sometimes it works fine and I believe the reason for it being that the ping pong is not able to be maintained because of the data transfer and from server side the client disconnects but client doesnt know that because of the prev data that is being shared and it never gets to know if any ping or pong was made.

I am just curious is there a way to log the ping and pong statements when they occur like rather than letting it happen automatically can we make it such that an event is called to work upon before transferring it to default implementation

**Comment by miguelgrinberg** [comment]
The ping/pong mechanism can be controlled using the `pingInterval` and `pingTimeout` options that you pass to the server. Beyond changing the timings there is no way to change anything unless you subclass the server and override this behavior.
